data/pigui/modules/station-view/06-shipRepairs.lua

```diff
@@ -3,15 +3,153 @@
 local StationView = import 'pigui/views/station-view'
+local ShipDef = import("ShipDef")
+local Game = import "Game"
+local Rand = import "Rand"
+local InfoFace = import 'ui/PiguiFace'
+local Format = import "Format"
+local Character = import "Character"
+local ModalWindow = import 'pigui/libs/modal-win.lua'
 local Lang = import 'Lang'
-
 local l = Lang.GetResource("ui-core")
 
+local rescaleVector = ui.rescaleUI(Vector2(1, 1), Vector2(1600, 900), true)
+
+local colors = ui.theme.colors
+local icons = ui.theme.icons
+
+local pionillium = ui.fonts.pionillium
+local orbiteer = ui.fonts.orbiteer
+
+local face = nil
+local stationSeed = false
+local damageToRepair = nil
+local repair_cost = 0
+
+local widgetSizes = ui.rescaleUI({
+	itemSpacing = Vector2(4, 9),
+	windowPadding = Vector2(14, 14),
+	faceSize = Vector2(586,565),
+	buttonSize = Vector2(100, 0),
+	verticalDummy = Vector2(0, 50),
+}, Vector2(1600, 900))
+
+
+local popup = ModalWindow.New('ChiefMechanicPopup', function(self)
+	ui.text(l.YOU_DONT_HAVE_ENOUGH_MONEY_FOR_THAT_OPTION)
+	ui.dummy(Vector2((ui.getContentRegion().x - 100*rescaleVector.x) / 2, 0))
+	ui.sameLine()
+	if ui.button(l.OK, Vector2(100*rescaleVector.x, 0)) then
+		self:close()
+	end
+end)
+
+
+local tryRepair = function (damage, price)
+	if Game.player:GetMoney() >= price then
+		Game.player:AddMoney(-price)
+		Game.player:SetHullPercent(Game.player:GetHullPercent() + damage)
+	else
+		popup:open()
+	end
+end
+
+
+local getRepairCost = function (percent, shipDef)
+	-- repairing 1% hull damage costs 0.1% of ship price
+	shipDef = shipDef or ShipDef[Game.player.shipId]
+	return math.ceil(shipDef.basePrice * (percent * 0.1)) * 0.01
+end
+
+
+local getRepairMessage = function (damage, price)
+	return string.interp(
+		l.REPAIR_X_HULL_DAMAGE_FOR_X, {
+			damage = string.format('%.1f', damage),
+			price = Format.Money(price)
+		})
+end
+
+
+function round(num, numDecimalPlaces)
+	local mult = 10^(numDecimalPlaces or 0)
+	return math.floor(num * mult + 0.5) / mult
+end
+
+
+local function drawShipRepair()
+	local hullPercent = round(Game.player:GetHullPercent())
+	local damage = 100 - hullPercent
+
+	local intro = string.interp(l.YOUR_HULL_IS_AT_X_INTEGRITY, {value = string.format('%.1f', hullPercent)})
+
+	ui.withStyleVars({WindowPadding = widgetSizes.windowPadding,
+		ItemSpacing = widgetSizes.itemSpacing}, function ()
+			local infoColumnWidth = ui.getContentRegion().x - widgetSizes.faceSize.x - widgetSizes.windowPadding.x*3
+
+			ui.child("ShipStatus", Vector2(infoColumnWidth, 0), {"AlwaysUseWindowPadding"}, function ()
+				if hullPercent > 99.9 then
+					ui.withFont(pionillium.medlarge.name, pionillium.medlarge.size,
+						function () ui.text(l.SHIP_IS_ALREADY_FULLY_REPAIRED) end)
+				else
+					ui.withFont(pionillium.medlarge.name, pionillium.medlarge.size, function ()
+						ui.text(getRepairMessage(damageToRepair, repair_cost))
+						damageToRepair = ui.sliderInt("", damageToRepair, 1, damage, "%d%%")
```

Comment:
Step 2: call pushItemWidth to set the sliderSize

```suggestion
						ui.pushItemWidth(widgetSizes.gaugeWidth)
						damageToRepair = ui.sliderInt("", damageToRepair, 1, damage, "%d%%")
```